packages/web-app/src/routes/appeal-details/application-decision-date.js

```diff
@@ -0,0 +1,22 @@
+const express = require('express');
+const applicationDecisionDateController = require('../../controllers/appeal-details/application-decision-date');
+const {
+  rules: decisionDateValidationRules,
+} = require('../../validation/appeal-details/decision-date');
+const combineDateInputsMiddleware = require('../../middleware/combine-date-inputs');
+const expressValidationErrorsToGovUkErrorList = require('../../lib/express-validation-errors-to-govuk-error-list');
+
+const router = express.Router();
+
+router.get('/', applicationDecisionDateController.getApplicationDecisionDate);
+router.post(
+  '/',
+  [
+    combineDateInputsMiddleware,
```

Comment:
Rather than calling `combineDateInputsMiddleware` as middleware so it needs to iterate all the form fields to check for a date field, couldn't it be a function which gets called when required for a specific date field, perhaps in the controller or wherever the date combining is needed?